### ISSUE TITLE
Weights are nan, loss is computed fine and gradients are too small

### ISSUE THREAD

**Issue by HannahMontanna** [tensorflow/tensorflow#7276]
I have been researching about this over the INTERNET for a very long time. But I still do not know where my problem lies. I am finetuning a siamese fashioned vgg. And the problem is that even after first iteration distance calcualted is fine, loss calculated is fine but the gradients are too small(which is a problem) and the weights that are updated are nan. I do not understand what the problem is. 
In the network, I have removed fully connected layers and the output from the last pooling layer is the input to the distance function. Here is my code. I have only attached important snippets of the code which I hope help understand the architecture. 
PS: stackoverflow was of no help that is why I am posting it here. 
PPS: I have attached snapshot of gradients as well
![screenshot from 2017-02-06 02-00-10](https://cloud.githubusercontent.com/assets/25512256/22630753/60a8760a-ec10-11e6-9a1f-d75ba573842b.png)

[code.txt](https://github.com/tensorflow/tensorflow/files/753375/code.txt)

**Comment by yaroslavvb** [comment]
BTW, this list is for bugs and feature requests in TensorFlow. It is common to get NaN's during normal operation. There was probably no response on stackoverflow because such problems are often due to a subtle bug in network design or training, and requires significant effort to track down. If you think this is a bug in TensorFlow, please more specific details of which functionality of TF is misbehaving

**Comment by HannahMontanna** [comment]
Makes sense. The problem is with the first two conv layers only. Latter ones are updated fine. Thank you for your reply, Ill take a look into design.
